src/factories/energy/createEnergyPlans.ts

```diff
@@ -44,9 +47,108 @@ Key values randomly allocated:
     public canCreateEnergyPlans(): boolean {return true}
 
 
-    public generateEnergyPlans(): EnergyPlanDetailV2[] | undefined  {
+    public generateEnergyPlans(): EnergyPlanDetail[] | EnergyPlanDetailV2[] | EnergyPlanDetailV3[] | undefined  {
+
+        var energyPlans: any;
+        if (this.detailVersion == 1) {
+            energyPlans = this.energyPlanDetail()
+        }
+        if (this.detailVersion == 3) {
```

Comment:
consider flipping those so it is ordered 3,2,1 and not 1,3,2. So it is consistent with the EnergyAccountDetailV4/3/2 similar pattern above.